src/cc/bcc_proc.c

```diff
@@ -82,6 +83,43 @@ int bcc_mapping_is_file_backed(const char *mapname) {
     STARTS_WITH(mapname, "[vsyscall]"));
 }
 
+/*
+Finds a file descriptor for a given inode if it's a memory-backed fd.
+*/
+char *bcc_procutils_memfd_path(int pid, uint64_t inum) {
+  char *path = NULL;
+  char *globstr;
+  char glob_buffer[1024];  // a ridiculous buffer size
+  int glob_size, i;
+  glob_t globbuf;
+  struct stat sb;
+
+  snprintf(glob_buffer, 1024, "/proc/%d/fd/*", pid);
+  glob_size = strlen(glob_buffer);
+
+  globstr = malloc(sizeof(char) * (glob_size + 1));
+  strcpy(globstr, glob_buffer);
```

Comment:
Can we use `opendir` / `readdir` instead of a glob search?

Reply:
I'll take a look :+1:

Reply:
done